src/core/net/mdns.cpp

```diff
@@ -2316,7 +2354,15 @@ void Core::ServiceEntry::AppendPtrRecordTo(TxMessage &aTxMessage, Section aSecti
     message = &aTxMessage.SelectMessageFor(aSection);
 
     ptr.Init();
-    ptr.SetTtl(ptrRecord.GetTtl());
+
+    if (aTxMessage.GetType() == TxMessage::kLegacyUnicastResponse)
+    {
+        ptr.SetTtl(Min(Core::RecordInfo::kMaxLegacyUnicastTtl, ptrRecord.GetTtl()));
+    }
+    else
+    {
+        ptr.SetTtl(ptrRecord.GetTtl());
+    }
```

Comment:
I was  thinking if we can make all these `SetTtl()` logic which is repeated possibly simpler?
- We can change `GetTtl()` so that it gets the `TxMessage::Type` as input and gets the TTL based on the type.
  - The challenge I see is that `TxMessage::Type` enum is defined after the `RecordInfo` class (so we may need to reorder stuff)
- Other way is just use a `bool aIsLegacyUnicast` instead?
- We can maybe do something similar and add a `bool aIsLegacyUnicast` to `UpdateCacheFlushFlagIn()` (to avoid `if()` checks?)

What do you think?